lumigator/python/mzai/backend/backend/tests/api/routes/test_datasets.py

```diff
@@ -83,6 +85,17 @@ def test_upload_delete(app_client: TestClient, valid_experiment_dataset: str):
     assert get_response.status_code == status.HTTP_404_NOT_FOUND
 
 
+def test_dataset_delete_error(app_client: TestClient):
```

Comment:
This is great as it can replicate the issue we had with unhandled exceptions, thanks!

Can we add something that tests the fix we have just applied? I'd expect that if we tried to delete a dataset that only appears in the DB and not on S3 we'd get a 204

Reply:
https://github.com/mozilla-ai/lumigator/pull/332/commits/b0c4d317ac77a99b3d1deb64bd22d220893da90f#diff-b92c2f9bf8c3f6c6a676575aea579fea1156b021c695ae70fcd005b1ef36a99d